apps/evm/lib/evm/refunds/sstore.ex

```diff
@@ -14,23 +14,31 @@ defmodule EVM.Refunds.Sstore do
   end
 
   defp basic_sstore_refund({key, new_value}, exec_env) do
-    case ExecEnv.get_storage(exec_env, key) do
-      {:ok, value} ->
-        if value != 0 && new_value == 0 do
-          @storage_refund
-        else
+    {updated_exec_env, current_value} = ExecEnv.storage(exec_env, key)
+
+    refund =
+      case current_value do
```

Comment:
Could `current_value` return an int? Then we could remove this case statement? i.e.

```ex
refund = if current_value != 0 && new_value == 0 do
  @storage_refund
else
  0
end
```